Scripts/script-FetchAnonymousInfo.yml

```diff
@@ -0,0 +1,115 @@
+commonfields:
+  id: FetchAnonymousInfo
+  version: -1
+name: FetchAnonymousInfo
+script: |-
+  import hashlib, json, datetime
+
+  INCIDENT_SAFE_FIELDS = [u'status', u'autime', u'isPlayground', u'activated', u'id', u'category',
+                          u'linkedIncidents', u'dueDate', u'version', u'openDuration', u'closed', u'reminder',
+                          u'type', u'parent', u'runStatus', u'occurred', u'phase', u'severity', u'account',
+                          u'created', u'modified', u'investigationId', u'playbookId', u'sla']
+
+  INCIDENT_HASH_VALUES = ['name', u'owner', 'closingUserId', 'closeNotes', 'reason', 'closeReason', 'CustomFields']
+
+
+  INDICATOR_SAFE_FIELDS = ['lastReputationRun', 'manualSetTime', 'timestamp', 'modified', 'calculatedTime',
+                           'indicator_type', 'source', 'version', 'investigationIDs', 'lastSeen', 'insightCache',
+                           'score', 'id', 'firstSeen', 'manualScore']
+
+  ANONYMOUES_DATA = (demisto.args()['anonymous'] == 'yes')
+
+  def hash_object(str_list_dict):
+      if not ANONYMOUES_DATA:
+          return str_list_dict
+      if str_list_dict == "" or str_list_dict is None:
+          return str_list_dict
+      if (type(str_list_dict)) == dict:
+          return dict(map(lambda (k,v): (k, hash_object(v)), str_list_dict.iteritems()))
+      if (type(str_list_dict) == list):
+          return map(hash_object, str_list_dict)
+
+      if (type(str_list_dict) in [str,unicode]):
+          str_value = str_list_dict.encode('utf-8')
+      else:
+          str_value = str(str_list_dict)
+      return hashlib.md5(str_value).hexdigest()
+
+  def parse_incident(incident):
+      incident_info = {key:incident[key] for key in INCIDENT_SAFE_FIELDS}
+      for field in INCIDENT_HASH_VALUES:
+          incident_info[field] = hash_object(incident[field])
+
+      if incident['labels'] is not None:
+          incident_info['labels'] = []
+          for label in incident['labels']:
+              incident_info['labels'].append({'type': label['type'], 'value': hash_object(label['value'])})
+      else:
+          incident_info['labels'] = None
```

Comment:
Do you have to set it if it doesn't exist?

Reply:
Not really, but I want it will have the same structure as the original incident object (and there the key exist)